Sources/NIO/CircularBuffer.swift

```diff
@@ -155,9 +155,17 @@ public struct CircularBuffer<E>: CustomStringConvertible, AppendableCollection {
 
     /// Removes all members from the circular buffer whist keeping the capacity.
     public mutating func removeAll(keepingCapacity: Bool = false) {
+        if keepingCapacity {
+            for index in 0..<self.count {
+                self.buffer[self.bufferIndex(ofIndex: index)] = nil
+            }
```

Comment:
Is there any reason we aren't just using `self.buffer.removeAll(keepingCapacity: true)`?

Reply:
@Lukasa yes, the ring buffer uses `self.buffer` as a pretty much constantly sized backing array. So an empty ring buffer with ring capacity 4 would look like this:

    [nil, nil, nil, nil]

if I were to call `self.buffer.removeAll(keepingCapacity: true)`, `self.buffer` were

    []

with some capacity reserved at the end of `self.buffer` for future expansions.

Basically it's `count` vs `capacity` in the backing store. Does that make sense?

Reply:
Yup, makes sense thanks.